### PR TITLE
Release 2.0

### PR DESCRIPTION
**Pull request by pepibumur**
### Short description 📝
This PR keeps track of the changes that will be part of the 2.0 release.

#### Resources
- [Project](https://github.com/tuist/tuist/projects/11)
- [2.0 deprecations](https://github.com/tuist/tuist/issues/2981)

#### Pull requests

- [x] [Remove CocoaPods's interface](https://github.com/tuist/tuist/pull/3369)
- [x] [Remove support for deprecated names of manifests](https://github.com/tuist/tuist/pull/3373)
- [ ] [Add EnvironmentVariable struct](https://github.com/tuist/tuist/pull/3320)
- [x] [Simplify TestAction's API](https://github.com/tuist/tuist/pull/3375)

### CODE REVIEW

**Comment by danyf90** [comment]
Converted to draft to avoid unwanted merge :D

**Comment by pepibumur** [comment]
Thanks @danyf90 for keeping this up to date.

File: projects/website/markdown/posts/2021-09-21-tuist-2.0/post.mdx [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+and in some scenarios,
+led to integration issues that surfaced at compilation time.
+Moreover,
+the integration of Swift Packages proposed by Apple proved to offer a poor developer experience.
+For example,
+the resolution of dependencies sometimes fails at launch time, 
+and it gets invalidated after deleting the derivd data directory.
+
+`Dependencies.swift` offered a standard solution across Carthage and Package dependencies
+and integrated dependencies into the projects' graphs to allow users to leverage existing Tuist like binary caching.
```
Comment (danyf90) [inline projects/website/markdown/posts/2021-09-21-tuist-2.0/post.mdx:104]: Should we mention that this has the advantage to also allow to use tuist caching on dependencies?

**Review by danyf90 — APPROVED** [review]

File: projects/website/markdown/posts/2021-09-21-tuist-2.0/post.mdx [hunk 2]
```diff
@@ -0,0 +1,148 @@
+---
+layout: post
+title: Releasing Tuist 2.0
+categories: ['Tuist', '2.0', 'Xcode', 'Swift', "Project generation"]
+excerpt: In this post, we share more details about this new major version of the project, 2.0, and present the direction we are taking as we move towards 3.0.
+author: pepibumur
+---
+
+After over [5 thousand commits](https://github.com/tuist/tuist/commits) from [120 contributors](https://github.com/tuist/tuist/graphs/contributors) and over three years after we landed the first commit on the project,
+I'm thrilled to share with all of you that we released the second major of the project, Tuist 2.0.
```
Comment (luispadron) [inline projects/website/markdown/posts/2021-09-21-tuist-2.0/post.mdx:10]:
```suggestion
I'm thrilled to share with all of you that we released the second major version of the project, Tuist 2.0.
```

File: projects/website/markdown/posts/2021-09-21-tuist-2.0/post.mdx [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+
+As you might know,
+projects that follow [semantic versioning](https://semver.org/) use major versions to flag breaking changes in the release.
+From the user standpoint, that can be seen as unfavorable because the migration is usually manual.
+However,
+breaking changes are often necessary to continue to improve the developer experience (DX). 
+In this new major iteration of Tuist,
+we're achieving that **by moving away from poor API designs**, 
+replacing some of them with **more straightforward and convenient APIs**,
+and **pruning features** that were distant from the project's direction.
```
Comment (luispadron) [inline projects/website/markdown/posts/2021-09-21-tuist-2.0/post.mdx:20]:
```suggestion
In this new major iteration of Tuist,
we're  **moving away from poor API designs**, 
replacing some of them with **more straightforward and convenient APIs**,
and **pruning features** that were distant from the project's direction.
```

File: projects/website/markdown/posts/2021-09-21-tuist-2.0/post.mdx [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+
+Because the release already provides release notes and migration guidelines,
```
Comment (luispadron) [inline projects/website/markdown/posts/2021-09-21-tuist-2.0/post.mdx:22]: Can we link to the notes / guide here?

File: projects/website/markdown/posts/2021-09-21-tuist-2.0/post.mdx [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+Therefore we'll overhaul our plugins architecture to piggyback on the Swift Package Manager's work.
+We are still fleshing out the details,
+but without spoiling you too much,
+plugins would be represented by Swift Packages that follow a convention defined by Tuist.
+For example,
+a task would become an executable in a Swift Package that follows the naming convention `tuist-{name}`.
+Thanks to that, 
+developers can add transitive dependencies to their plugins,
+and we can leverage the Swift Package Manager CLI to export the plugins in a distributable format.
+
```
Comment (luispadron) [inline projects/website/markdown/posts/2021-09-21-tuist-2.0/post.mdx:47]:
```suggestion

Checkout a more in depth RFC: [here](https://github.com/tuist/tuist/discussions/3491)
```

**Review by luispadron — APPROVED** [review]
🚀

File: projects/website/markdown/posts/2021-09-21-tuist-2.0/post.mdx (hunk 1 above)
Comment (pepibumur) [reply projects/website/markdown/posts/2021-09-21-tuist-2.0/post.mdx:105]: Yes. We should definitively mention that.

File: projects/website/markdown/posts/2021-09-21-tuist-2.0/post.mdx (hunk 4 above)
Comment (pepibumur) [reply projects/website/markdown/posts/2021-09-21-tuist-2.0/post.mdx:22]: Yes! I was waiting for that part to be written to have a hyperlink I can refer to.

**Pull request merged**